modules/cli/src/main/scala/scala/cli/commands/Bsp.scala

```diff
@@ -1,33 +1,43 @@
 package scala.cli.commands
 
 import caseapp._
+import upickle.default._
 
 import scala.build.Build
 import scala.build.bsp.BspThreads
+import scala.build.options.BuildOptions
 import scala.concurrent.Await
 import scala.concurrent.duration.Duration
+import scala.io.Source
 
 object Bsp extends ScalaCommand[BspOptions] {
   def run(options: BspOptions, args: RemainingArgs): Unit = {
 
     if (options.shared.logging.verbosity >= 3)
       pprint.stderr.log(args)
 
-    val buildOptions     = options.buildOptions
-    val bloopRifleConfig = options.shared.bloopRifleConfig()
-    val logger           = options.shared.logging.logger
+    val sharedOptions: SharedOptions =
+      options.jsonOptions.map { optionsPath =>
+        val source = Source.fromFile(optionsPath.toFile())
+        try read[SharedOptions](source.mkString)
+        finally source.close()
```

Comment:
We can use os-lib here, like
```suggestion
        val source = os.read(os.Path(optionsPath, os.pwd))
        read[SharedOptions](source)
```